### ISSUE TITLE
ObjectVersion.size() return is Integer not long

### ISSUE THREAD

**Issue by erichelgeson** [aws/aws-sdk-java-v2#1299]
## Expected Behavior

When retrieving `ObjectVersion` - it should use the appropriate data type that can hold the size attribute. 

## Current Behavior

I receive an `ObjectVersion` with a large size I get - ```java.lang.NumberFormatException: For input string: "2484911842"``` as it is larger than `Integer.MAX_VALUE`

Full stacktrace:
```
Caused by: java.lang.NumberFormatException: For input string: "2484911842"
    at java.base/java.lang.NumberFormatException.forInputString(NumberFormatException.java:65)
    at java.base/java.lang.Integer.parseInt(Integer.java:652)
    at java.base/java.lang.Integer.parseInt(Integer.java:770)
    at software.amazon.awssdk.protocols.core.StringToValueConverter$SimpleStringToValue.convert(StringToValueConverter.java:58)
    at software.amazon.awssdk.protocols.xml.internal.unmarshall.XmlPayloadUnmarshaller$SimpleTypePayloadUnmarshaller.unmarshall(XmlPayloadUnmarshaller.java:118)
    at software.amazon.awssdk.protocols.xml.internal.unmarshall.XmlProtocolUnmarshaller.unmarshall(XmlProtocolUnmarshaller.java:86)
    at software.amazon.awssdk.protocols.xml.internal.unmarshall.XmlPayloadUnmarshaller.unmarshallSdkPojo(XmlPayloadUnmarshaller.java:55)
    at software.amazon.awssdk.protocols.xml.internal.unmarshall.XmlPayloadUnmarshaller.lambda$unmarshallList$0(XmlPayloadUnmarshaller.java:65)
    at java.base/java.util.ArrayList.forEach(ArrayList.java:1540)
    at software.amazon.awssdk.protocols.xml.internal.unmarshall.XmlPayloadUnmarshaller.unmarshallList(XmlPayloadUnmarshaller.java:62)
    at software.amazon.awssdk.protocols.xml.internal.unmarshall.XmlProtocolUnmarshaller.unmarshall(XmlProtocolUnmarshaller.java:86)
    at software.amazon.awssdk.protocols.xml.internal.unmarshall.XmlProtocolUnmarshaller.unmarshall(XmlProtocolUnmarshaller.java:74)
    at software.amazon.awssdk.protocols.xml.internal.unmarshall.XmlProtocolUnmarshaller.unmarshall(XmlProtocolUnmarshaller.java:58)
    at software.amazon.awssdk.protocols.xml.internal.unmarshall.AwsXmlResponseHandler.unmarshallResponse(AwsXmlResponseHandler.java:82)
    at software.amazon.awssdk.protocols.xml.internal.unmarshall.AwsXmlResponseHandler.handle(AwsXmlResponseHandler.java:61)
    at software.amazon.awssdk.protocols.xml.internal.unmarshall.AwsXmlResponseHandler.handle(AwsXmlResponseHandler.java:41)
    at software.amazon.awssdk.awscore.client.handler.AwsSyncClientHandler$Crc32ValidationResponseHandler.handle(AwsSyncClientHandler.java:88)
    at software.amazon.awssdk.core.client.handler.BaseClientHandler.lambda$interceptorCalling$2(BaseClientHandler.java:151)
    at software.amazon.awssdk.core.client.handler.AttachHttpMetadataResponseHandler.handle(AttachHttpMetadataResponseHandler.java:40)
    at software.amazon.awssdk.core.client.handler.AttachHttpMetadataResponseHandler.handle(AttachHttpMetadataResponseHandler.java:28)
    at software.amazon.awssdk.core.internal.http.pipeline.stages.HandleResponseStage.handleSuccessResponse(HandleResponseStage.java:89)
    ... 64 more
```

## Possible Solution

Seems very similar to #917 - May need to add a customization for the `ObjectVersion` in `services/s3/src/main/resources/codegen-resources/customization.config`

You may want to review all usages of the Shape size in s3 codegen.

## Steps to Reproduce (for bugs)

Create a file larger than `Integer.MAX_VALUE` in an s3 bucket.
Call the `listObjectVersions` method.

```
s3Client.listObjectVersions(ListObjectVersionsRequest.builder().bucket(bucketName).prefix(prefix).versionIdMarker(versionIdMarker).build())
```

## Context

Cannot retrieve Object Versions.

## Your Environment

* AWS Java SDK version used: 2.5.29
* JDK version used: 11.0.3
* Operating System and version: macOS 10.13.6

**Comment by zoewangg** [comment]
Thank you for reporting! I am able to reproduce the issue. You are correct; we'd need to add customization for that shape and we will look through all the size shapes in `s3`.

**Comment by zoewangg** [comment]
Fixed via #1306 and it has been released as part of `2.6.0`. Can you try with the latest version?  Feel free to reopen if you have further questions.
